src/common/download.h

```diff
@@ -32,5 +32,11 @@
 
 namespace tools
 {
-  bool download(const std::string &path, const std::string &url);
+  typedef void *download_async_handle;
```

Comment:
I should also mention that implicit detachment is deprecated with `boost::thread`, and was never allowed for `std::thread`. So I would try to make sure that `join` or `detach` is called somewhere before the thread is destroyed for long-term consistency.